sparta/sparta/collection/Collectable.hpp

```diff
@@ -303,10 +303,11 @@ namespace sparta{
                 sparta_assert(pipeline_col_ != nullptr,
                             "Collectables can only added to PipelineCollectors... for now");
 
-                if(collect && !prev_annot_.empty()) {
+                if(collect) {
```

Comment:
Have you checked this in Argos? I just want to double check that this doesn't cause every collected location to have a gigantic record stretching from time = 0.

Reply:
How I can check that?

Reply:
Run a sim, dump a DB, open it up and see whether the output makes sense. If every location has a record at the very beginning then something is probably wrong.

Reply:
Actually, would it make more sense to create a single open/closed empty record?

Reply:
It looks like your changes to the reader are already enough to properly handle an empty DB, so I don't really see why the empty record is necessary.

Reply:
Well, if someone collects a pipeout for one component in simulation (which never changes its value or even starts with one), and runs their entire simulation (1000s of cycles), the database will be empty instead of just blank for that component.  Not sure what the right behavior is.

Reply:
Hm, that's a good point. I think in that case the index file would still have heartbeats, so we could probably use that to distinguish a corrupt DB from an empty one.

Reply:
I don't feel comfortable with this change.  There isn't enough testing to see if I broke something downstream, and I barely remember why I originally wrote the code this way.  I'm going to revert the change to the Collectable, but keep the rest.